sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ReplaceNullWithFalseInPredicateSuite.scala

```diff
@@ -200,13 +200,15 @@ class ReplaceNullWithFalseInPredicateSuite extends PlanTest {
 
   test("inability to replace null in non-boolean values of CaseWhen") {
     val nestedCaseWhen = CaseWhen(
```

Comment:
Those tests are modified and one branch is added in `CaseWhen` to avoid new rule converting `CaseWhen` to `If` for better test coverage.